[RHCLOUD-19197] feature: "provisioning" app type

MikelAlejoBR opened the pull request:
They are starting to integrate with us and they need this application
type.

## Links

[[RHCLOUD-19197]](https://issues.redhat.com/browse/RHCLOUD-19197)

lindgrenj6 commented on dao/seeds/source_types.yml:
```diff
@@ -66,6 +66,24 @@ amazon:
+    - type: provisioning-arn
+      name: Provisioning ARN
+      fields:
+        - component: text-field
+          name: authentication.authtype
+          hideField: true
+          initializeOnMount: true
+          initialValue: provisioning-arn
+        - name: authentication.username
+          component: text-field
+          label: ARN
+          isRequired: true
+          validate:
+            - type: required
+            - type: pattern
+              pattern: "^arnaws:.*"
+            - type: min-length
+              threshold: 10
```
These just need moved 2 spaces to the left - fields is an array of objects.

MikelAlejoBR replied:
I've reduced the indentation in two spaces. I was confused by this, though:

> fields is an array of objects.

I think that these are equivalent?

```yaml
array:
  - one
  - two
  - three
```

```yaml
array:
- one
- two
- three
```

lindgrenj6 replied:
They might be treated the same by some parsers but not by others - thats where it gets fun. For example parsing the second one in ruby works, but the first one fails. For the k8s Go library it works, and for gopkg.in/yaml v2 & v3 it works.


tl;dr: yaml sucks. but the whitespace usually denotes an object rather than an array.

MikelAlejoBR replied:
Wow, TIL. [Yaml Spec V1.2 §10.1.1.2.](https://yaml.org/spec/1.2.2/#10112-generic-sequence)

MikelAlejoBR committed:
feature: application type for the Provisioning service

They have requested a new application type to make their integration
efforts easier.

ezr-ondrej commented on dao/seeds/application_types.yml:
```diff
@@ -69,3 +69,11 @@
     satellite:
     - receptor_node
+"/insights/platform/provisioning-service":
+  display_name: Provisioning service
```
I'd rather go with Provisioning only, but unimportat ATM :)
```suggestion
  display_name: Provisioning
```

lzap commented on dao/seeds/source_types.yml:
```diff
@@ -66,6 +66,24 @@ amazon:
         - type: min-length
           threshold: 10
+    - type: provisioning-service-arn
+      name: Provisioning service's ARN
```
Just not to get this skipped, it's also here:

```suggestion
      name: Provisioning's ARN
```
